packages/types/src/interfaces/contractsAbi/definitions.ts

```diff
@@ -0,0 +1,174 @@
+    len: 'u32',
+    cellsPerElem: 'u64',
+    layout: 'InkStorageLayout'
+  },
+  InkLayoutStruct: {
+    fields: 'Vec<InkLayoutStructField>'
+  },
+  InkLayoutStructField: {
+    layout: 'InkStorageLayout',
+    name: 'Text'
+  },
+  InkDiscriminant: 'u32',
+  InkLayoutEnum: {
+    dispatchKey: 'InkLayoutKey',
+    variants: 'BTreeMap<InkDiscriminant, InkLayoutStruct>'
+  }
+};
+
+const spec = {
+  InkPath: 'Vec<Text>',
+  InkDisplayName: 'InkPath',
+  InkConstructorSpec: {
+    name: 'Text',
+    selector: 'InkSelector',
+    args: 'Vec<InkMessageParamSpec>',
+    docs: 'Vec<Text>'
+  },
+  InkContractSpec: {
+    constructors: 'Vec<InkConstructorSpec>',
+    messages: 'Vec<InkMessageSpec>',
+    events: 'Vec<InkEventSpec>',
+    docs: 'Vec<Text>'
+  },
+  InkEventParamSpec: {
+    name: 'Text',
+    indexed: 'bool',
+    type: 'InkTypeSpec',
+    docs: 'Vec<Text>'
+  },
+  InkEventSpec: {
+    name: 'Text',
+    args: 'Vec<InkEventParamSpec>',
+    docs: 'Vec<Text>'
+  },
+  InkMessageParamSpec: {
+    name: 'Text',
+    type: 'InkTypeSpec'
+  },
+  InkMessageSpec: {
+    name: 'Text',
+    selector: 'InkSelector',
+    mutates: 'bool',
+    payable: 'bool',
+    args: 'Vec<InkMessageParamSpec>',
+    returnType: 'Option<InkTypeSpec>',
+    docs: 'Vec<Text>'
+  },
+  InkSelector: '[u8; 4]',
+  InkTypeSpec: {
+    type: 'MtLookupTypeId',
+    displayName: 'InkDisplayName'
+  }
+};
+
+const registry = {
+  MtField: {
+    name: 'Option<Text>',
+    type: 'MtLookupTypeId'
+  },
+  MtLookupTypeId: 'u32',
+  MtType: {
+    def: 'MtTypeDef',
+    params: 'Vec<MtLookupTypeId>',
+    path: 'Vec<Text>'
+  },
+  MtTypeDef: {
+    _enum: {
+      Array: 'MtTypeDefArray',
+      Composite: 'MtTypeDefComposite',
+      Primitive: 'MtTypeDefPrimitive',
+      Sequence: 'MtTypeDefSequence',
+      Slice: 'MtTypeDefSlice',
+      Tuple: 'MtTypeDefTuple',
+      Variant: 'MtTypeDefVariant'
+    }
+  },
+  MtTypeDefArray: {
+    len: 'u16',
+    type: 'MtLookupTypeId'
+  },
+  MtTypeDefComposite: {
+    fields: 'Vec<MtField>'
+  },
+  MtTypeDefPrimitive: {
+    // this enum definition is mapped in api-contracts/inkTypes.ts
+    _enum: ['Bool', 'Char', 'Str', 'U8', 'U16', 'U32', 'U64', 'U128', 'I8', 'I16', 'I32', 'I64', 'I128']
+  },
+  MtTypeDefSequence: {
+    type: 'MtLookupTypeId'
+  },
+  MtTypeDefSlice: {
+    type: 'MtLookupTypeId'
+  },
+  MtTypeDefTuple: 'Vec<MtLookupTypeId>',
+  MtTypeDefVariant: {
+    variants: 'Vec<MtVariant>'
+  },
+  MtVariant: {
+    discriminant: 'Option<u64>',
+    fields: 'Vec<MtField>',
+    name: 'Text'
+  }
+};
+
+export default {
+  rpc: {},
+  types: {
+    ...layout,
+    ...registry,
+    ...spec,
+    InkContractContract: {
+      authors: 'Vec<Text>',
+      name: 'Text',
+      version: 'Text'
+    },
+    InkContractSource: {
+      compiler: 'Text',
+      hash: '[u8; 32]',
+      language: 'Text'
+    },
+    InkProject: {
+      contract: 'InkContractContract',
+      metadata_version: 'Text',
```

Comment:
This is still incorrect although marked as resolved, see https://github.com/polkadot-js/api/pull/1662/files#diff-db27842186eb038cb3e29484bc7e5888R147-R156